comps/retrievers/langchain/vdms/ingest.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+#
+
+import io
+import os
+import time
+
+import numpy as np
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain_community.embeddings import HuggingFaceBgeEmbeddings, HuggingFaceEmbeddings, HuggingFaceHubEmbeddings
+from langchain_community.vectorstores import VDMS
+from langchain_community.vectorstores.vdms import VDMS_Client
+from PIL import Image
+from vdms_config import (  # , INDEX_SCHEMA, VDMS_URL
+    DISTANCE_STRATEGY,
+    EMBED_MODEL,
+    INDEX_NAME,
+    SEARCH_ENGINE,
+    VDMS_HOST,
+    VDMS_PORT,
+)
+
+tei_embedding_endpoint = os.getenv("TEI_EMBEDDING_ENDPOINT")
+client = VDMS_Client(VDMS_HOST, VDMS_PORT)
+
+timeout = 180  # Timeout in seconds
+check_interval = 5  # Check every 5 seconds
+
+
+def check_embedding_endpoint(embedder):
+    try:
+        test_sentence = "embedder available test."
+        embedder.embed_documents([test_sentence])
+        return True
+    except Exception as e:
+        print(f"Error embedder is unavailable: {e}")
+        return False
+
+
+def pdf_loader(file_path):
+    try:
+        import easyocr
+        import fitz
+    except ImportError:
+        raise ImportError(
+            "`PyMuPDF` or 'easyocr' package is not found, please install it with "
+            "`pip install pymupdf or pip install easyocr.`"
+        )
+
+    doc = fitz.open(file_path)
+    reader = easyocr.Reader(["en"])
+    result = ""
+    for i in range(doc.page_count):
+        page = doc.load_page(i)
+        pagetext = page.get_text().strip()
+        if pagetext:
+            result = result + pagetext
+        if len(doc.get_page_images(i)) > 0:
+            for img in doc.get_page_images(i):
+                if img:
+                    pageimg = ""
+                    xref = img[0]
+                    img_data = doc.extract_image(xref)
+                    img_bytes = img_data["image"]
+                    pil_image = Image.open(io.BytesIO(img_bytes))
+                    img = np.array(pil_image)
+                    img_result = reader.readtext(img, paragraph=True, detail=0)
+                    pageimg = pageimg + ", ".join(img_result).strip()
+                    if pageimg.endswith("!") or pageimg.endswith("?") or pageimg.endswith("."):
+                        pass
+                    else:
+                        pageimg = pageimg + "."
+                result = result + pageimg
+    return result
+
+
+def ingest_documents():
+    """Ingest PDF to VDMS from the data/ directory that
+    contains Edgar 10k filings data for Nike."""
+    # Load list of pdfs
+    company_name = "Nike"
+    data_path = "data/"
```

Comment:
in most cases, this path will result in an authorization issue?

Reply:
The ingest.py file will be removed from the retriever code.